addons/huntir/functions/fnc_handleFired.sqf

```diff
@@ -33,15 +33,20 @@ if (_ammo != "F_HuntIR") exitWith {};
         _huntir setPosATL _position;
         [{
             EXPLODE_1_PVT(_this select 0,_huntir);
-            private ["_deltaT"];
             if (isNull _huntir) exitWith {
                 [_this select 1] call CBA_fnc_removePerFrameHandler;
             };
+            private ["_deltaT", "_velocity"];
+            _velocity = velocity _huntir;
             if (damage _huntir > 0) then {
                 _deltaT = ACE_time - (_huntir getVariable [QGVAR(lastTime), ACE_time]);
-                _huntir setVelocity (velocity _huntir vectorAdd [0, 0, -9.8066 * (damage _huntir) * _deltaT]);
+                _velocity = _velocity vectorAdd [0, 0, -9.8066 * (damage _huntir) * _deltaT];
```

Comment:
But only the `else` block overrides the velocity. The `if` block velocity change is relative to the current speed.